lib/tape.js

```diff
@@ -0,0 +1,259 @@
+
+	notOk: 'falsy',
+	false: 'falsy',
+	notok: 'falsy',
+
+	error: 'ifError',
+	ifErr: 'ifError',
+	iferror: 'ifError',
+
+	equal: 'is',
+	equals: 'is',
+	isEqual: 'is',
+	strictEqual: 'is',
+	strictEquals: 'is',
+
+	notEqual: 'not',
+	notStrictEqual: 'not',
+	notStrictEquals: 'not',
+	isNotEqual: 'not',
+	doesNotEqual: 'not',
+	isInequal: 'not',
+
+	deepEqual: 'deepEqual',
+	isEquivalent: 'deepEqual',
+	same: 'deepEqual',
+
+	notDeepEqual: 'notDeepEqual',
+	notEquivalent: 'notDeepEqual',
+	notDeeply: 'notDeepEqual',
+	notSame: 'notDeepEqual',
+	isNotDeepEqual: 'notDeepEqual',
+	isNotEquivalent: 'notDeepEqual',
+	isInequivalent: 'notDeepEqual',
+
+	skip: 'skip',
+	throws: 'throws',
+	doesNotThrow: 'notThrows'
+};
+
+const unsupportedTestFunctions = new Set([
+	// No equivalent in AVA:
+	'timeoutAfter',
+	'deepLooseEqual',
+	'looseEqual',
+	'looseEquals',
+	'notDeepLooseEqual',
+	'notLooseEqual',
+	'notLooseEquals'
+]);
+
+/**
+ * Updates CommonJS and import statements from Tape to AVA
+ * @return string with test function name if transformations were made
+ */
+function updateTapeRequireAndImport(j, ast) {
+	let testFunctionName = null;
+	ast.find(j.CallExpression, {
+		callee: {name: 'require'},
+		arguments: arg => arg[0].value === 'tape'
+	})
+	.filter(p => p.value.arguments.length === 1)
+	.forEach(p => {
+		p.node.arguments[0].value = 'ava';
+		testFunctionName = p.parentPath.value.id.name;
+	});
+
+	ast.find(j.ImportDeclaration, {
+		source: {
+			type: 'Literal',
+			value: 'tape'
+		}
+	})
+	.forEach(p => {
+		p.node.source.value = 'ava';
+		testFunctionName = p.value.specifiers[0].local.name;
+	});
+
+	return testFunctionName;
+}
+
+export default function tapeToAva(fileInfo, api) {
+	const j = api.jscodeshift;
+	const ast = j(fileInfo.source);
+
+	const testFunctionName = updateTapeRequireAndImport(j, ast);
+
+	if (testFunctionName) {
+		const transforms = [
+			function detectUnsupportedNaming() {
+				// Currently we only support "t" as the test argument name
+				const validateTestArgument = p => {
+					const lastArg = p.value.arguments[p.value.arguments.length - 1];
+					if (lastArg && lastArg.params && lastArg.params[0]) {
+						const lastArgName = lastArg.params[0].name;
+						if (lastArgName !== 't') {
+							throw new Error(`Argument to test function should be named "t" not "${lastArgName}"`);
+						}
+					}
+				};
+
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: testFunctionName}
+					}
+				})
+				.forEach(validateTestArgument);
+
+				ast.find(j.CallExpression, {
+					callee: {name: testFunctionName}
+				})
+				.forEach(validateTestArgument);
+			},
+
+			function detectUnsupportedFeatures() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: ({name}) => unsupportedTestFunctions.has(name)
+					}
+				})
+				.forEach(p => {
+					throw new Error(`Codemod transformation of "${p.value.callee.property.name}" is not supported`);
```

Comment:
Do we want to fail the transformation when unsupported features are found? I think it might be more helpful to log out `file.path` and the line/column info, and `console.warn` so the user knows to go make an update.